NitroxPatcher/Patches/Dynamic/CyclopsSonarButton_Update_Patch.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NitroxClient.GameLogic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic
+{
+    /// <summary>
+    /// Prevents sonar from turning off automatically for the player that isn't currently piloting the Cyclops.
+    /// </summary>
+    public class CyclopsSonarButton_Update_Patch : NitroxPatch, IDynamicPatch
+    {
+        public static readonly MethodInfo TARGET_METHOD = Reflect.Method((CyclopsSonarButton t) => t.Update());
+
+        private static readonly OpCode INJECTION_OPCODE = OpCodes.Ldsfld;
+        private static readonly object INJECTION_OPERAND = Reflect.Field(() => Player.main);
+
+        private static NitroxId currentCyclopsId;
+
+        public static void Prefix(CyclopsSonarButton __instance)
+        {
+            currentCyclopsId = NitroxEntity.GetId(__instance.subRoot.gameObject);
+        }
+
+        public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+            Validate.NotNull(INJECTION_OPERAND);
+            
+            /* Normally in the Update()
+             * if (Player.main.GetMode() == Player.Mode.Normal && this.sonarActive)
+		     * {
+		     *     this.TurnOffSonar();
+		     * }
+             * this part will be changed into:
+             * if (CyclopsSonarButton_Update_Patch.ShouldTurnOff() && Player.main.GetMode() == Player.Mode.Normal && this.sonarActive)
+             */
+            List<CodeInstruction> codeInstructions = new List<CodeInstruction>(instructions);
```

Comment:
Is there a reason you casting to List? Afaik you van iterate a IEnumerable with foreach.

Reply:
There's a moment I need to get the code instruction that's 2 lines further, that's why I don't use a foreach